android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -739,15 +739,21 @@ namespace android
   // Fills mapobject's metadata from UserMark
   void Framework::InjectMetadata(JNIEnv * env, jclass const clazz, jobject const mapObject, UserMark const * userMark)
   {
-    feature::Metadata metadata;
+    using feature::Metadata;
+
+    Metadata metadata;
     frm()->FindClosestPOIMetadata(userMark->GetOrg(), metadata);
 
     static jmethodID const addId = env->GetMethodID(clazz, "addMetadata", "(ILjava/lang/String;)V");
     ASSERT ( addId, () );
 
-    for (feature::Metadata::EType t : metadata.GetPresentTypes())
+    for (Metadata::EType const t : metadata.GetPresentTypes())
     {
-      jstring metaString = jni::ToJavaString(env, metadata.Get(t));
+      // TODO: It is not a good idea to pass raw strings to UI. Calling separate getters should be a better way.
+      // Upcoming change: how to pass opening hours (parsed) into Editor's UI? How to get edited changes back?
+      jstring metaString = t == Metadata::FMD_WIKIPEDIA ?
```

Comment:
Duplicated lines are both marked with TODO and will be removed after refactoring in another PR. I don't see any good solution to avoid duplication now in a better way.